torch/_subclasses/fake_tensor.py

```diff
@@ -552,6 +553,12 @@ def index_tensor(fake_mode, func, *args, **kwargs):
 # takes in multiple-devices, dont default to default device handling
 @register_op_impl(aten.index_put.default)
 def index_put(fake_mode, func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    check_same_device(
+        new_kwargs["input"], new_kwargs["values"], allow_cpu_scalar_tensors=True
+    )
```

Comment:
I don't think this is right -- it looks like we allow values to be a CUDA scalar Tensor as well:
```
x = torch.tensor([1., 2.])
y = torch.tensor(2., device='cuda')
idx = torch.tensor(0)
torch.index_put(x, y, idx)
```

Reply:
No, it doesn't do device error checking. it is opting out of that